src/components/content/catalog/services/unregister/UnregisterService.tsx

```diff
@@ -27,6 +38,51 @@ function UnregisterService({
         },
     });
 
+    const unregister = () => {
+        unregisterRequest.mutate();
+    };
+
+    if (unregisterRequest.isSuccess) {
+        let refreshedServiceName = '';
+        let refreshedServiceCsp = '';
+        categoryOclData.forEach((serviceList, serviceName) => {
+            if (currentServiceName.split('@')[0] === serviceName) {
+                const versionMapper = getVersionMapper(serviceName, serviceList);
+                versionMapper.forEach((versionList, version) => {
+                    if (serviceName + '@' + version === currentServiceName) {
+                        refreshedServiceName = currentServiceName;
+                        refreshedServiceCsp = versionList[0].csp.valueOf();
+                        return false;
+                    }
+                });
+                if (refreshedServiceName.length === 0) {
+                    const firstVersionList: ServiceTemplateDetailVo[] = versionMapper.values().next()
+                        .value as ServiceTemplateDetailVo[];
+                    if (firstVersionList.length > 0) {
+                        refreshedServiceName = serviceName + '@' + firstVersionList[0].version;
+                        refreshedServiceCsp = firstVersionList[0].csp.valueOf();
+                    }
+                }
+                return;
+            } else {
+                const firstServiceList: ServiceTemplateDetailVo[] = categoryOclData.values().next()
+                    .value as ServiceTemplateDetailVo[];
+                if (firstServiceList.length > 0) {
+                    refreshedServiceName = serviceName + '@' + firstServiceList[0].version;
+                    refreshedServiceCsp = firstServiceList[0].csp.valueOf();
+                }
+            }
+        });
+        navigate({
+            pathname: catalogPageRoute,
+            hash: '#' + category,
+            search: createSearchParams({
+                currentServiceNameVersion: refreshedServiceName,
```

Comment:
as mentioned on chat, please make the query to have all 4 parameters separately. 

serviceName, version, csp and hosting type